tools/integration_tests/util/client/storage_client.go

```diff
@@ -97,3 +102,47 @@ func CreateObjectOnGCS(ctx context.Context, client *storage.Client, object, cont
 
 	return nil
 }
+
+func CreateStorageClientWithTimeOut(ctx *context.Context, storageClient **storage.Client, time time.Duration, t *testing.T) func() {
+	var err error
+	var cancel context.CancelFunc
+	*ctx, cancel = context.WithTimeout(*ctx, time)
+	*storageClient, err = CreateStorageClient(*ctx)
+	if err != nil {
+		log.Fatalf("client.CreateStorageClient: %v", err)
+	}
+	// return func to close storage client and release resources.
+	return func() {
+		err := (*storageClient).Close()
+		if err != nil {
+			t.Log("Failed to close storage client")
```

Comment:
We should get rid of t *testing.T from the utility. In that way, this would be a generic method.

We can do that, return error from this method as well as the return function should also return an error.